### ISSUE TITLE
Torch v1.12 fixes bug to train a model, but YoloV5 rejects the version.

### ISSUE THREAD

**Issue by Exortions** [ultralytics/yolov5#8609]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar bug report.

### YOLOv5 Component

Training

### Bug

Recently, training in YOLOv5 would crash because of an error in PyTorch. (See this issue: https://github.com/pytorch/pytorch/issues/74016)

PyTorch released a fix in version 1.12, but YOLOv5 doesn't train with this error:
requirements: torch!=1.12.0,>=1.7.0 not found and is required by YOLOv5, attempting auto-update...
It then downgrades PyTorch to v1.11.0, which once again crashes the training process when ran with it.

How can I get PyTorch v1.12 to run with YOLOv5?

### Environment

- YOLO: YOLOv5 latest
- OS: Windows 11 (WSL)
- Python: 3.9.0

### Minimal Reproducible Example

```
git clone https://github.com/ultralytics/yolov5
cd yolov5

pip install torch==1.12

python train.py --img 640 --batch 16 --epochs 10 --data coco128.yaml --weights yolov5s.pt --cache
```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by equal-l2** [comment]
It seems that they intentionally exclude torch 1.12.0 because the version has a serious bug in multi-GPU environments.
See https://github.com/ultralytics/yolov5/issues/8395 for the detail.

TL; DR: The only thing we users can do is wait for the new release of torch (1.12.1, presumably).

**Comment by glenn-jocher** [comment]
@Exortions @equal-l2 good news 😃! Your original issue may now be fixed ✅ in PR #8621. torch==1.12.0 installs are now allowed, and an assert torch!=1.12.0 has been inserted into a new smart_DDP() function. The Docker image is using torch nightly and trains DDP correctly.

To receive this update:
- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!

**Comment by Exortions** [comment]
Thank you so much!

**Comment by glenn-jocher** [comment]
@Exortions you're very welcome! 🎉 Kudos to the persevering and passionate YOLO community - and the incredible Ultralytics team who made this update possible. Please don't hesitate to reach out if you have any more questions or need further assistance. Good luck with your YOLOv5 adventures!
